ftrackhq/ftrack-javascript
# feat: Allow aborting uploads using AbortSignal

Resolves FT-4fe346cb-3f20-438d-9b07-d1ee95f19e33

- [x] The PR title is suitable as a release note

I branched this off the latest tag as I had some issues testing with the new event hub client implementation.

## Changes

* Deprecate `xhr` option as that is will not be compatible with multi-part uploads
* Introduce a signal option for aborting uploads (similar to other methods)

## Test

1. Check that you agree with API
2. Verify that tests run.
